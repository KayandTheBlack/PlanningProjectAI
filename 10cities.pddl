(define (problem autogen)
  (:domain worldsEndAndBeyond)

  (:objects
    Origin cityA cityB cityC cityD cityE cityF cityG cityH cityI cityJ - city
    hotelA_A hotelA_B hotelA_C hotelB_A hotelB_B hotelB_C hotelC_A hotelC_B hotelC_C hotelD_A hotelD_B hotelD_C hotelE_A hotelE_B hotelE_C hotelF_A hotelF_B hotelF_C hotelG_A hotelG_B hotelG_C hotelH_A hotelH_B hotelH_C hotelI_A hotelI_B hotelI_C hotelJ_A hotelJ_B hotelJ_C - hotel
  )

  (:init
    (currentLocation Origin)

    (hotelAt hotelA_A cityA)
    (hotelAt hotelA_B cityA)
    (hotelAt hotelA_C cityA)
    (hotelAt hotelB_A cityB)
    (hotelAt hotelB_B cityB)
    (hotelAt hotelB_C cityB)
    (hotelAt hotelC_A cityC)
    (hotelAt hotelC_B cityC)
    (hotelAt hotelC_C cityC)
    (hotelAt hotelD_A cityD)
    (hotelAt hotelD_B cityD)
    (hotelAt hotelD_C cityD)
    (hotelAt hotelE_A cityE)
    (hotelAt hotelE_B cityE)
    (hotelAt hotelE_C cityE)
    (hotelAt hotelF_A cityF)
    (hotelAt hotelF_B cityF)
    (hotelAt hotelF_C cityF)
    (hotelAt hotelG_A cityG)
    (hotelAt hotelG_B cityG)
    (hotelAt hotelG_C cityG)
    (hotelAt hotelH_A cityH)
    (hotelAt hotelH_B cityH)
    (hotelAt hotelH_C cityH)
    (hotelAt hotelI_A cityI)
    (hotelAt hotelI_B cityI)
    (hotelAt hotelI_C cityI)
    (hotelAt hotelJ_A cityJ)
    (hotelAt hotelJ_B cityJ)
    (hotelAt hotelJ_C cityJ)

    (flight Origin cityA)
    (flight Origin cityB)
    (flight Origin cityC)
    (flight Origin cityD)
    (flight Origin cityE)
    (flight Origin cityF)
    (flight Origin cityG)
    (flight Origin cityH)
    (flight Origin cityI)
    (flight Origin cityJ)
    (flight cityA cityB)
    (flight cityA cityF)
    (flight cityA cityH)
    (flight cityB cityC)
    (flight cityB cityG)
    (flight cityB cityI)
    (flight cityC cityA)
    (flight cityC cityB)
    (flight cityC cityG)
    (flight cityD cityA)
    (flight cityD cityC)
    (flight cityD cityJ)
    (flight cityE cityC)
    (flight cityE cityD)
    (flight cityE cityI)
    (flight cityF cityD)
    (flight cityF cityE)
    (flight cityF cityJ)
    (flight cityG cityA)
    (flight cityG cityF)
    (flight cityG cityH)
    (flight cityH cityA)
    (flight cityH cityC)
    (flight cityH cityI)
    (flight cityI cityA)
    (flight cityI cityD)
    (flight cityI cityJ)
    (flight cityJ cityE)
    (flight cityJ cityF)
    (flight cityJ cityH)

    (= (citiesVisited) 0)
    (= (totalDays) 0)
    (= (minDaysInCity) 1)
    (= (maxDaysInCity) 3)

    (= (stayDuration Origin) 0)
    (= (stayDuration cityA) 0)
    (= (stayDuration cityB) 0)
    (= (stayDuration cityC) 0)
    (= (stayDuration cityD) 0)
    (= (stayDuration cityE) 0)
    (= (stayDuration cityF) 0)
    (= (stayDuration cityG) 0)
    (= (stayDuration cityH) 0)
    (= (stayDuration cityI) 0)
    (= (stayDuration cityJ) 0)
    (= (totalcost) 0)
    (= (hotelcost hotelA_A) 86)
    (= (hotelcost hotelA_B) 69)
    (= (hotelcost hotelA_C) 119)
    (= (hotelcost hotelB_A) 216)
    (= (hotelcost hotelB_B) 128)
    (= (hotelcost hotelB_C) 191)
    (= (hotelcost hotelC_A) 205)
    (= (hotelcost hotelC_B) 50)
    (= (hotelcost hotelC_C) 143)
    (= (hotelcost hotelD_A) 125)
    (= (hotelcost hotelD_B) 65)
    (= (hotelcost hotelD_C) 85)
    (= (hotelcost hotelE_A) 61)
    (= (hotelcost hotelE_B) 60)
    (= (hotelcost hotelE_C) 71)
    (= (hotelcost hotelF_A) 226)
    (= (hotelcost hotelF_B) 100)
    (= (hotelcost hotelF_C) 197)
    (= (hotelcost hotelG_A) 209)
    (= (hotelcost hotelG_B) 59)
    (= (hotelcost hotelG_C) 77)
    (= (hotelcost hotelH_A) 121)
    (= (hotelcost hotelH_B) 163)
    (= (hotelcost hotelH_C) 98)
    (= (hotelcost hotelI_A) 104)
    (= (hotelcost hotelI_B) 185)
    (= (hotelcost hotelI_C) 227)
    (= (hotelcost hotelJ_A) 219)
    (= (hotelcost hotelJ_B) 122)
    (= (hotelcost hotelJ_C) 169)

    (= (flightcost Origin cityA) 0)
    (= (flightcost Origin cityB) 0)
    (= (flightcost Origin cityC) 0)
    (= (flightcost Origin cityD) 0)
    (= (flightcost Origin cityE) 0)
    (= (flightcost Origin cityF) 0)
    (= (flightcost Origin cityG) 0)
    (= (flightcost Origin cityH) 0)
    (= (flightcost Origin cityI) 0)
    (= (flightcost Origin cityJ) 0)
    (= (flightcost cityA cityB) 196)
    (= (flightcost cityA cityF) 108)
    (= (flightcost cityA cityH) 138)
    (= (flightcost cityB cityC) 165)
    (= (flightcost cityB cityG) 127)
    (= (flightcost cityB cityI) 168)
    (= (flightcost cityC cityA) 106)
    (= (flightcost cityC cityB) 134)
    (= (flightcost cityC cityG) 121)
    (= (flightcost cityD cityA) 151)
    (= (flightcost cityD cityC) 162)
    (= (flightcost cityD cityJ) 188)
    (= (flightcost cityE cityC) 139)
    (= (flightcost cityE cityD) 173)
    (= (flightcost cityE cityI) 199)
    (= (flightcost cityF cityD) 160)
    (= (flightcost cityF cityE) 101)
    (= (flightcost cityF cityJ) 101)
    (= (flightcost cityG cityA) 107)
    (= (flightcost cityG cityF) 112)
    (= (flightcost cityG cityH) 110)
    (= (flightcost cityH cityA) 187)
    (= (flightcost cityH cityC) 136)
    (= (flightcost cityH cityI) 175)
    (= (flightcost cityI cityA) 187)
    (= (flightcost cityI cityD) 190)
    (= (flightcost cityI cityJ) 162)
    (= (flightcost cityJ cityE) 164)
    (= (flightcost cityJ cityF) 111)
    (= (flightcost cityJ cityH) 135)

  )

  (:goal
    (and
      (>= (citiesVisited) 10)
      (>= (totalDays) 20)
      (>= (totalcost) 100)
      (<= (totalcost) 1000000)
    )
  )

  (:metric 
    minimize (totalcost)
  )

)
