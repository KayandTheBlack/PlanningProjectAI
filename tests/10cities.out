
ff: parsing domain file
domain 'WORLDSENDANDBEYOND' defined
 ... done.
ff: parsing problem file
problem 'AUTOGEN' defined
 ... done.


metric established (normalized to minimize): ((1.00*[RF0](TOTALCOST)) - () + 0.00)

checking for cyclic := effects --- OK.

ff: search configuration is  best-first on 1*g(s) + 5*h(s) where
    metric is ((1.00*[RF0](TOTALCOST)) - () + 0.00)

advancing to distance:   20
                         19
                         18
                         17
                         16
                         15
                         14
                         13
                         12
                         11
                         10
                          9
                          8
                          7
                          6
                          5
                          4
                          3
                          2
                          1
                          0

ff: found legal plan as follows

step    0: FLYANDSTAY ORIGIN CITYF HOTELF_B
        1: INCREASESTAY CITYF HOTELF_B
        2: INCREASESTAY CITYF HOTELF_B
        3: FLYANDSTAY CITYF CITYE HOTELE_B
        4: FLYANDSTAY CITYE CITYC HOTELC_B
        5: FLYANDSTAY CITYC CITYG HOTELG_B
        6: INCREASESTAY CITYC HOTELC_B
        7: INCREASESTAY CITYG HOTELG_B
        8: INCREASESTAY CITYE HOTELE_B
        9: INCREASESTAY CITYE HOTELE_B
       10: INCREASESTAY CITYG HOTELG_B
       11: INCREASESTAY CITYC HOTELC_B
       12: FLYANDSTAY CITYG CITYH HOTELH_C
       13: FLYANDSTAY CITYH CITYA HOTELA_B
       14: FLYANDSTAY CITYA CITYB HOTELB_B
       15: INCREASESTAY CITYA HOTELA_B
       16: FLYANDSTAY CITYB CITYI HOTELI_A
       17: INCREASESTAY CITYA HOTELA_B
       18: FLYANDSTAY CITYI CITYD HOTELD_B
       19: FLYANDSTAY CITYD CITYJ HOTELJ_B
     

time spent:    0.00 seconds instantiating 150 easy, 0 hard action templates
               0.00 seconds reachability analysis, yielding 62 facts and 150 actions
               0.00 seconds creating final representation with 61 relevant facts, 25 relevant fluents
               0.00 seconds computing LNF
               0.00 seconds building connectivity graph
               0.18 seconds searching, evaluating 7386 states, to a max depth of 0
               0.18 seconds total time

